common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java

```diff
@@ -374,6 +376,32 @@ public int removeBlocks(String appId, String execId, String[] blockIds) {
       .collect(Collectors.toMap(Pair::getKey, Pair::getValue));
   }
 
+  /**
+   * Diagnose the possible cause of the shuffle data corruption by verify the shuffle checksums
+   */
+  public Cause diagnoseShuffleBlockCorruption(
+      String appId,
+      String execId,
+      int shuffleId,
+      long mapId,
+      int reduceId,
+      long checksumByReader) {
+    ExecutorShuffleInfo executor = executors.get(new AppExecId(appId, execId));
+    String fileName = "shuffle_" + shuffleId + "_" + mapId + "_0.checksum";
+    File probeFile = ExecutorDiskUtils.getFile(
+      executor.localDirs,
+      executor.subDirsPerLocalDir,
+      fileName);
+    File parentFile = probeFile.getParentFile();
+
+    File[] checksumFiles = parentFile.listFiles(f -> f.getName().startsWith(fileName));
+    assert checksumFiles.length == 1;
```

Comment:
Btw, do we want an assertion here ?
If there is some reason why the checksum is not present (for example, cleanup) - why not simply return `Cause.UNKNOWN_ISSUE` in that case ?

Reply:
> Is it because it was decided to append the checksum algorithm to the file name and the exact name of the file is unknown?

Exactly. I have switched to use `Files.newDirectoryStream` instead.

Reply:
> If there is some reason why the checksum is not present (for example, cleanup) - why not simply return Cause.UNKNOWN_ISSUE in that case ?

Good idea, addressed.